src/input/reader_to_storage.cc

```diff
@@ -502,6 +573,31 @@ StorageBase * ReaderToStorage::make_storage_from_default(const string &dflt_str,
 
 
 
+StorageBase * ReaderToStorage::make_transposed_storage(PathBase &p, const Type::TypeBase *type) {
+	ASSERT(try_transpose_read_, "Unset flag try_transpose_read_!\n");
+
+	int arr_size = p.get_array_size();
+	if ( arr_size == 0 ) {
+		THROW( ExcInputError() << EI_Specification("Empty array during transposition.")
+			<< EI_ErrorAddress(p.as_string()) << EI_InputType(type->desc()) );
+	} else if ( arr_size > 0 ) {
+		if (transpose_index_ == 0) transpose_array_sizes_.push_back( arr_size );
+		p.down(transpose_index_);
+		StorageBase *storage = make_storage(p, type);
+		p.up();
+		return storage;
+	} else { // arr_size == -1
+		 THROW( ExcInputError()
```

Comment:
If this method is used only for p.is_array_type() true, this case should not happen. Thus we should have an ASSERT here.